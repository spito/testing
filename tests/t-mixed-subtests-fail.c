#include "cut.h"

TEST(prvni) {
    SUBTEST("trololo") {
        DEBUG_MSG("%d", SUBTEST_NO);
    }

    REPEATED_SUBTEST("x", 5) {
        DEBUG_MSG("%d", SUBTEST_NO);
    }

    SUBTEST("chacha") {
        ASSERT(0);
    }
}

TEST(druhy) {
    REPEATED_SUBTEST("x", 2) {
        DEBUG_MSG("%d", SUBTEST_NO);
    }
 
    SUBTEST("trololo") {
        DEBUG_MSG("%d", SUBTEST_NO);
    }

    REPEATED_SUBTEST("x", 5) {
        DEBUG_MSG("%d", SUBTEST_NO);
    }
}

TEST(treti) {
    REPEATED_SUBTEST("fail", -1) {
        DEBUG_MSG("upsi");
    }
}

TEST(ctvrty) {
    REPEATED_SUBTEST("x", 2) {
        DEBUG_MSG("alfa %d", SUBTEST_NO);
    }

    REPEATED_SUBTEST("x", 2) {
        DEBUG_MSG("%d beta", SUBTEST_NO);
    }
}