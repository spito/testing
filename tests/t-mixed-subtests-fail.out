[  1] prvni: 7 subtests
    trololo..................................................................OK
        debug messages:
          1 (mixed-subtests-fail.c:5)

    x........................................................................OK
        debug messages:
          1 (mixed-subtests-fail.c:9)

    x........................................................................OK
        debug messages:
          2 (mixed-subtests-fail.c:9)

    x........................................................................OK
        debug messages:
          3 (mixed-subtests-fail.c:9)

    x........................................................................OK
        debug messages:
          4 (mixed-subtests-fail.c:9)

    x........................................................................OK
        debug messages:
          5 (mixed-subtests-fail.c:9)

    chacha.................................................................FAIL
        assert '0' (mixed-subtests-fail.c:13)

[  1] prvni (overall)......................................................FAIL
[  2] druhy: 8 subtests
    x........................................................................OK
        debug messages:
          1 (mixed-subtests-fail.c:19)

    x........................................................................OK
        debug messages:
          2 (mixed-subtests-fail.c:19)

    trololo..................................................................OK
        debug messages:
          1 (mixed-subtests-fail.c:23)

    x........................................................................OK
        debug messages:
          1 (mixed-subtests-fail.c:27)

    x........................................................................OK
        debug messages:
          2 (mixed-subtests-fail.c:27)

    x........................................................................OK
        debug messages:
          3 (mixed-subtests-fail.c:27)

    x........................................................................OK
        debug messages:
          4 (mixed-subtests-fail.c:27)

    x........................................................................OK
        debug messages:
          5 (mixed-subtests-fail.c:27)

[  2] druhy (overall)........................................................OK
[  3] treti................................................................FAIL
    assert '0 < -1' (mixed-subtests-fail.c:32)

[  4] ctvrty: 4 subtests
    x #1.....................................................................OK
        debug messages:
          alfa 1 (mixed-subtests-fail.c:39)

      #2.....................................................................OK
        debug messages:
          alfa 2 (mixed-subtests-fail.c:39)

      #3.....................................................................OK
        debug messages:
          1 beta (mixed-subtests-fail.c:43)

      #4.....................................................................OK
        debug messages:
          2 beta (mixed-subtests-fail.c:43)

[  4] ctvrty (overall).......................................................OK

Summary:
  tests:          4
  succeeded:      2
  filtered out:   0
  suppressed:     0
  skipped:        0
  failed:         2
